src/transactions/feed/TransactionFeedV2.tsx

```diff
@@ -0,0 +1,196 @@
+import React, { useEffect, useMemo, useState } from 'react'
+import { ActivityIndicator, SectionList, StyleSheet, View } from 'react-native'
+import SectionHead from 'src/components/SectionHead'
+import GetStarted from 'src/home/GetStarted'
+import { useSelector } from 'src/redux/hooks'
+import { getFeatureGate, getMultichainFeatures } from 'src/statsig'
+import { StatsigFeatureGates } from 'src/statsig/types'
+import colors from 'src/styles/colors'
+import { Spacing } from 'src/styles/styles'
+import NoActivity from 'src/transactions/NoActivity'
+import { useTransactionFeedV2Query } from 'src/transactions/api'
+import EarnFeedItem from 'src/transactions/feed/EarnFeedItem'
+import NftFeedItem from 'src/transactions/feed/NftFeedItem'
+import SwapFeedItem from 'src/transactions/feed/SwapFeedItem'
+import TokenApprovalFeedItem from 'src/transactions/feed/TokenApprovalFeedItem'
+import TransferFeedItem from 'src/transactions/feed/TransferFeedItem'
+import { pendingStandbyTransactionsSelector } from 'src/transactions/reducer'
+import { NetworkId, type TokenTransaction } from 'src/transactions/types'
+import { groupFeedItemsInSections } from 'src/transactions/utils'
+import { walletAddressSelector } from 'src/web3/selectors'
+
+type PaginatedData = {
+  [timestamp: number]: TokenTransaction[]
```

Comment:
Should we type it like this? In case we read it with a timestamp that wasn't added?
```suggestion
  [timestamp: number]: TokenTransaction[] | undefined
```

Reply:
@jeanregisser I still plan to introduce the data persistence (with some workarounds) to have that instant first page load as we have on the existing transaction feed. But I agree - this should include `undefined` while the persistence is absent.

Or alternatively, instead of this:
```
 const [paginatedData, setPaginatedData] = useState<PaginatedData>({})
 ```
 I can define it like this:
 ```
  const [paginatedData, setPaginatedData] = useState<PaginatedData>({ [FIRST_PAGE_TIMESTAMP]: [] })
  ```
  
The first page will always be refreshed anyways so it can be empty from the start.